crates/cheatnet/src/state.rs

```diff
@@ -222,6 +262,19 @@ impl CheatnetState {
     pub fn get_cheated_caller_address(&self, address: &ContractAddress) -> Option<ContractAddress> {
         get_cheat_for_contract(&self.global_prank, &self.pranked_contracts, address)
     }
+
+    pub fn add_new_call_and_update_current_call(&mut self, entry_point: CallEntryPoint) {
```

Comment:
This vaguely describes what is really done, which is like "registering" nested call
I'd call this `enter_nested_call` or `register_nested_call` and the code describes the rest of the behaviour

Reply:
`enter_nested_call` is a great name.

Reply:
https://github.com/foundry-rs/starknet-foundry/pull/1539/commits/4278f33601ddee3ddb8001b47d00ea33a14ee39b